testsuite/MDAnalysisTests/coordinates/test_writer_api.py

```diff
@@ -52,10 +52,9 @@ def test_ts_error(writer, tmpdir):
         fn = str(tmpdir.join('out.xtc'))
     elif writer == mda.coordinates.LAMMPS.DATAWriter:
         pytest.skip("DATAWriter requires integer atom types")
-    elif writer == mda.coordinates.H5MD.H5MDWriter:
-        if not mda.coordinates.H5MD.HAS_H5PY:
-            pytest.skip("H5MDWriter requires h5py to be installed, "
-                        "fails MIN dependency test")
+    elif writer == mda.coordinates.H5MD.H5MDWriter and not mda.coordinates.H5MD.HAS_H5PY:
+        pytest.skip("H5MDWriter requires h5py to be installed, "
+                    "fails MIN dependency test")
```

Comment:
Also, remove 3.6, this might change in the future. ... and the ubuntu/macOS
```suggestion
        pytest.skip("skipping H5MDWriter test because h5py is not installed")
```